specs/labels-and-localization/README.md

```diff
@@ -0,0 +1,190 @@
+- Browse to that resx file on disk, and manually copy it into the app project directory, in `Resources\`
+- Add it to the Blazor project (as Build Action = Embedded Resource). Then, open it in the resx editor, and set Access Modifier = Public or Internal (to generate the `.designer.cs` file)
+- To translate/localize the resources, clone the resx file (e.g. `NimbleLabels.fr.resx`) and translate the values in it. The app also needs to follow the [Microsoft Blazor localization docs](https://learn.microsoft.com/en-us/aspnet/core/blazor/globalization-localization?view=aspnetcore-7.0&pivots=server) in terms of opting into localization, configuring the supported locales, etc.
+- To consume the resources, in the Razor file using the NimbleThemeProvider, add:
+```cs
+[Inject]
+internal IStringLocalizer<NimbleLabels>? LabelProvider { get; set; }
+```
+Then bind to the theme provider labels in the Razor template:
+```xml
+<NimbleThemeProvider Theme="Theme" LabelNumberFieldDecrement="@LabelProvider!["labelNumberFieldDecrement"]" LabelNumberFieldIncrement="@LabelProvider!["labelNumberFieldIncrement"]">
+```
+
+The process above isn't great. When updated versions of Nimble Blazor are released, if they have new labels, the consuming app would need to repeat the process of manually copying the resx, and translating it. They'd also need to manually add the new label properties in their Razor file using the NimbleThemeProvider.
+
+If we have any clients that will be using Nimble Blazor and non-English locales we should probably do additional research to see if we can come up with a more seamless approach.
+
+*Possible Alternatives:*
+- Add an `IStringLocalizer` property to the `NimbleThemeProvider` Razor component, and codegen the `Label*` properties to use it if it's set. This would improve the process of telling Nimble about the localized resources, but wouldn't change the process of manually copying the resx to start with.
+- Define an MSBuild task that copies the resx file, similar to what's outlined in [this GitHub comment](https://github.com/ni/nimble/issues/558#issuecomment-1129279985). Not much better / still a manual process.
+
+### Plan for Client-Provided Labels
+Examples: Button content, menu item content, `nimble-banner` `dismissButtonLabel`, `nimble-table` column `action-menu-label`
+
+Almost all of these labels are context-specific, i.e. different buttons on the page will have different text, and different columns will have different action menu labels.
+
+Clients will localize those labels themselves:
+- For Angular, they can use the `localize` function and/or `i18n` attributes/ attribute prefixes
+- For Blazor, they can use `.resx` files and `IStringLocalizer`
+
+The banner's `dismissButtonLabel` will be semi-redundant once we have a Nimble-provided label for banner dismiss buttons, so we could consider removing that attribute in the future.
+
+## Alternative Implementations / Designs
+
+Alternative options for Nimble-provided labels:
+### Slots
```

Comment:
Kinda related but this alternative had me wondering how does Google translate work for WebVIs. If you look at a running WebVI in chrome, right click on the page, and choose translate, a bunch of the content translates pretty well.

Was wondering how it chooses what to translate and landed on this: https://www.w3.org/International/questions/qa-translate-flag

My takeaway there is that Google translate probably only sees the contents of slots, and specific attributes, but not custom attributes.

Wonder if we did something like
```
<nimble-theme-provider>
<nimble-i18n>
<span nimble-label="increment-button">
Increment
</span>
```
And used a children directive to watch for changes and textContent of the span, could we actually have these strings participate in Google translate, etc.

Reply:
It's possible things already participate well with Google translate since ultimately we target native elements in shadow roots (a, button, input, etc). I didn't check how our elements currently behave.

Reply:
For the most part, Google Translate seems to handle our components fine (I ran it on the Angular and example apps).
 
I also ran it on [the Chromatic site for my prototype branch](https://60e89457a987cf003efc0a5b-ydpjhmvyhr.chromatic.com/) and mostly confirmed what you said:
- custom attributes are not translated. So `label-number-field-decrement` on the theme provider wouldn't get translated. So `fromView` wouldn't matter in that case (since it wouldn't be translated if it was reflected as an attribute or not). However,
- If we use the localized labels in "standard" ways (button content, inside spans, as slotted content, etc etc), Google Translate will still translate that content fine. So the labels on the theme provider aren't translated, but their eventual usages in the components would be.

Given that switching to the span approach / monitoring text content and all that seems like overkill.

Reply:
Added a bullet point about Google Translate in the main proposal, and added the rest of this to the Alternatives section